docs/architecture.md

```diff
@@ -1,42 +1,104 @@
-is included in Elemental is [{{elemental.operator.name}}]({{elemental.operator.url}}) which
-is responsible for managing OS upgrades and managing a secure device inventory to assist
+The {{elemental.operator.name}} is responsible for managing OS upgrades and managing a secure device inventory to assist
 with zero touch provisioning.
+It provides an {{elemental.operator.name}} Helm Chart and an {{elemental.register.name}}.
+
+### {{elemental.operator.name}} Helm Chart
+The {{elemental.operator.name}} Helm Chart must be installed on a Rancher Cluster. It enables new hosts to:
+
+- register against the {{elemental.operator.name}}.
+- retrieve the Elemental installation configuration required for the Elemental OS installation from Kubernetes resources.
+- download and install the [{{ranchersystemagent.name}}]({{ranchersystemagent.url}}), which enables Rancher to provision and manage K3s and RKE2 on the Elemental nodes.
+
+The {{elemental.operator.name}} allows control of the Elemental Nodes by extending the Kubernetes APIs with a set of _elemental.cattle.io_ [Kubernetes CRDs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/):
+
+- MachineRegistration
+- MachineInventory
+- MachineInventorySelector
+- MachineInventorySelectorTemplate
+- ManagedOSImage
+- ManagedOSVersion
+- ManagedOSVersionChannel
+
+#### MachineRegistration
+A MachineRegistration resource should be created for each new host to be provisioned by Elemental.
+The MachineRegistration includes the Elemental installation configuration (provided by the user) and a registration token (generated by the {{elemental.operator.name}}), from which a unique registration URL is derived.
+
+The registration URL is the way through which an host can access the {{elemental.operator.name}} services, to kick off the Elemental provisioning process.
+
+The MachineRegistration has a `Ready` condition which turns to true when the {{elemental.operator.name}} has successfully generated the registration URL and an associated ServiceAccount. From this point on the target host can connect to the registration URL to kick off the provisioning process.
+
+An HTTP GET request against the registration URL returns the _registration file_: a .yaml file containing the registration data (i.e., the _spec:config:elemental:registration_ section only from the just created MachineRegistration).
+The registration file contains all the required data to allow the target host to perform self registration and start the Elemental provisioning. See the [config:elemental:registration section in the MachineRegistration reference](machineregistration-reference#configelementalregistration)) for more details on the available registration options.
+
+
+#### MachineInventory
+When a new host registers successfully, the {{elemental.operator.name}} creates a MachineInventory resource representing that particular host.
```

Comment:
We should note that it's created by host agent and not by operator running on the management cluster